force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,11 +174,45 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
+        // Try to find the account
+        Account account;
+        try {
+            account = [SELECT Id FROM Account WHERE Name = :accountName LIMIT 1];
+        } catch (QueryException e) {
+            account = null;
+        }
+    
+        // If account doesn't exist, create it
+        if (account == null) {
+            account = new Account(Name = accountName);
+            upsert account;
```

Comment:
You dont really need "upsert" here. The only way to get here would be if account is null and you created a new account and thus only "insert" should ever happen here.